price-estimator/src/filter.rs

```diff
@@ -157,6 +183,24 @@ async fn estimate_amounts_at_price<T>(
     Ok(warp::reply::json(&result))
 }
 
+async fn estimate_best_ask_price<T>(
+    token_pair: TokenPair,
+    _query: QueryParameters,
+    price_rounding_buffer: f64,
+    orderbook: Arc<Orderbook<T>>,
+) -> Result<impl warp::Reply, Infallible> {
+    let price = orderbook
+        .get_pricegraph()
+        .await
+        .estimate_exchange_rate(token_pair.into(), 0.0)
+        .map(|xrate| {
+            // NOTE: Exchange rate is the inverse of price for an ask order.
+            1.0 / apply_rounding_buffer(xrate, price_rounding_buffer)
+        });
+
+    Ok(warp::reply::json(&price))
```

Comment:
We should probably use a type from `models` here so it's clear what the schema is. Currently price is an `Option<f64>`.

Reply:
Let me know if this is what you meant.

Reply:
That's fine. We could still think about whether we always want this to be a json object but most importantly I wanted to see the type.

Reply:
A number is valid JSON though. We can modify the schema if needed.